### ISSUE TITLE
[4.1][Bug] Issue with select2 and IsColumnNullable - wrong model instance

### ISSUE THREAD

**Issue by warton** [Laravel-Backpack/CRUD#2842]
# Bug report

### What I did

I've created field of type `select2` on column that can't be nullable.

### What I expected to happen

I expected select to do not allow choose empty option.

### What happened

First option was `-` and I could choose it and even submit the form.

### What I've already tried to fix it

I fixed it inside `select2.blade.php` field template changing row 4:

```PHP
$entity_model = $field['model']();
```

to

```PHP
$entity_model = $crud->model;
```

I did also a fix, but I do not have rights to push my branch and create PR.

### Backpack, Laravel, PHP, DB version

When I run ```php artisan backpack:version``` the output is:
### PHP VERSION:
PHP 7.4.4 (cli) (built: Mar 17 2020 13:49:19) ( ZTS Visual C++ 2017 x64 )
Copyright (c) The PHP Group
Zend Engine v3.4.0, Copyright (c) Zend Technologies
    with Xdebug v2.8.1, Copyright (c) 2002-2019, by Derick Rethans

### LARAVEL VERSION:
v7.11.0@f4563bd2e0875c59a1f7967abdbe5cef7f240117

### BACKPACK VERSION:
4.1.5@e2c166b065f6b405747db5a5f4d36a2a88f8dc22

**Comment by pxpm** [comment]
Indeed @warton .

Thanks for spotting, this is indeed failing because we check $field['name'] against the connected model, and not the current model. I will submit a fix for it. 

Best

**Comment by tabacitu** [comment]
Fixed by @pxpm 's #2854 - thanks a lot @warton ! It's merged and tagged so a `composer update backpack/crud` should do it for you.
